src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/DesigntimeLicenseContextSerializer.cs

```diff
@@ -14,6 +14,19 @@ namespace System.ComponentModel.Design
     /// </summary>
     public class DesigntimeLicenseContextSerializer
     {
+        internal const string s_enableUnsafeBinaryFormatterInDesigntimeLicenseContextSerialization = "System.ComponentModel.TypeConverter.EnableUnsafeBinaryFormatterInDesigntimeLicenseContextSerialization";
+        private const string s_enableBinaryFormatter = "System.Runtime.Serialization.EnableUnsafeBinaryFormatterSerialization";
+
+        private static bool EnableUnsafeBinaryFormatterInDesigntimeLicenseContextSerialization
+        {
+            get => AppContext.TryGetSwitch(s_enableUnsafeBinaryFormatterInDesigntimeLicenseContextSerialization, out bool isEnabled) ? isEnabled : false;
```

Comment:
This is different than how I suggested with `AllowCustomResourceTypes`. The difference is that this is a property which will always query the AppContext every time it is read.

With `AllowCustomResourceTypes`, it is a property over a backing compiler generated field, which is initialized a single time:

[https://github.com/dotnet/runtime/blob/7ab65423373f04998f3d365d564b3fc0e55fdf2b/src/libraries/System.Private.CoreLib/src/System/Resources/ResourceReader.cs#L131](https://github.com/dotnet/runtime/blob/7ab65423373f04998f3d365d564b3fc0e55fdf2b/src/libraries/System.Private.CoreLib/src/System/Resources/ResourceReader.cs#L131)